actors/product-actor.ts

```diff
@@ -0,0 +1,35 @@
+import { type Page } from '@playwright/test';
+import { ProductRegistrationPage } from '../pages/product-registration-page';
+import { ProductPage } from '../pages/product-page';
+
+export class ProductActor {
+    readonly page: Page;
+    readonly productPage: ProductPage;
+    readonly productRegistrationPage: ProductRegistrationPage;
+
+    constructor(page: Page) {
+        this.page = page;
+        this.productRegistrationPage = new ProductRegistrationPage(page);
+        this.productPage = new ProductPage(page)
+    }
+    
+    async handleProductRegistration() {
+        const actions = Object.freeze({
+            setRegister: Symbol("register"),
+            setNoRegister: Symbol("not require register"),
+        });
+
+        let action = await Promise.any([
+            this.productPage.registerButton.waitFor().then(() => actions.setRegister),
+            this.productPage.registrationNotRequired.waitFor().then(() => actions.setNoRegister),
+        ]);
+
+        if (action === actions.setRegister) {
```

Comment:
Regarding the conditions in the code, yes, for now it looks simple, but I think, if we have 4 different products, we will end-up with conditions all over the place. It does not matter if it's in an actor, in the page or in the test. 
At the moment what I can think of is just to have different tests for each product. Even if use sub-typing for pages or actors, we still would need different tests or conditions in the tests to determine what objects to import.
So , well I guess we can live with this for now until we figure out something better.

Reply:
I agree with, for more products or situations we need think more about how to deal with for future. Thanks.